script.js

```diff
@@ -1 +1,348 @@
+    rowElement.classList.add("row");
+    // set each square
+    // j is the column number
+    for (let j = 0; j < row.length; j += 1) {
+      // one square element
+      const square = document.createElement("div");
+      square.classList.add("square");
+      if (j === row.length - 1) {
+        square.classList.add("right");
+      }
+      if (j === 0) {
+        square.classList.add("left");
+      }
+      if (i === 0) {
+        square.classList.add("top");
+      }
+      if (i === board.length - 1) {
+        square.classList.add("bottom");
+      }
+      // set the text of the square according to the array
+      square.innerText = board[i][j];
+      rowElement.appendChild(square);
+      // set the click all over again
+      // eslint-disable-next-line
+      square.addEventListener("click", () => {
+        squareClick(i, j);
+      });
+    }
+    // add a single row to the board
+    boardContainer.appendChild(rowElement);
+  }
+};
+//check empty cells
+const emptyCellCheck = () => {
+  emptyCells = [];
+  for (let i = 0; i < board.length; i++) {
+  for (let j = 0; j <board.length; j++) {
+    if (board[i][j] === "") {
+        emptyCells.push([i,j]);
+    }
+}}
+return emptyCells;
+}
+
+
+// find available corner Moves
+const emptyCornerCheck = () => {
+  cornerMove = [];
+  if (board[0][0] ==="") {
+    cornerMove.push([0,0]);
+  }
+  if (board[0][board.length-1] ==="") {
+    cornerMove.push([0,board.length-1]);
+  }
+    if (board[board.length-1][0] ==="") {
+    cornerMove.push([board.length-1,0]);
+  }
+  if (board[board.length-1][board.length-1] ==="") {
+    cornerMove.push([board.length-1,board.length-1]);
+  }
+  return cornerMove;
+}
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+const getRandomIndex = (max) => {
+  return Math.floor(Math.random() * max);
+};
+
+/*#############
+GAMEPLAY LOGIC
+##############*/
+
+// switch the global values from one player to the next
+const togglePlayer = (isComp) => {
+  emptyCellCheck();
+  emptyCornerCheck();
+  gameInfo.innerText = "";
+  if (checkWin()===true) {
+      gameInfo.innerText = `${currentPlayer} is winner`;
+      lockBoard = true;
+    }
+  if (gameTie() === false) {
+    if (currentPlayer === huPlayer) {
+    currentPlayer = aiPlayer;
+    isComp ? aiPlay() : null;
+  } else {
+    currentPlayer = huPlayer;
+  }
+  }
+  else {
+    gameInfo.innerText = "It's a tie";
+    lockBoard = true;
+  } 
+};
+
+// computer choose randomly
+const computerPlay = (anArray) => {
+  let randomIndex = getRandomIndex(anArray.length);
+  setTimeout (function () {
+    squareClick(anArray[randomIndex][0],anArray[randomIndex][1]);
+  },500);
+}
+
+const computerPick = () => {
+  //computer pick the coordinate that will either win the game or block X
+  if (winMove.length > 0) {
+    //console.log("win move is played");
+    computerPlay(winMove);
+  }
+  if (blockMove.length > 0 && winMove.length ===0) {
```

Comment:
Just food for thought, I see the statement `winMove.length ===0` and `blockMove.length === 0` etc are repeated a lot of times, it might be a good idea to just create some nested `if else` statements and put the repeated conditionals inside. This way, you can write less code, and it also improves readability. It will not affect performance.